src/charm.py

```diff
@@ -212,6 +243,8 @@ def _on_update_status(self, _):
             )
             return
 
+        self.upgrade._reconcile_upgrade(event)
```

Comment:
Same as the other PR, this is the spec: 
```
All events except stop 
If a unit’s state is restarting, it will:
  * Check compatibility with versions
    * If incompatible: set unit status to BlockedStatus(“Upgrade incompatible. Rollback to previous revision with `juju refresh`”) and stop further execution
 * (Start services, if applicable, and) perform health checks
   * If checks fail: set unit status and stop further executions
 * Set its state to healthy
```